src/helpers/HookConfigLib.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {HookConfig, ModuleEntity} from "../interfaces/IModuleManager.sol";
+
+// Hook types:
+// Exec hook: bools for hasPre, hasPost
+// Validation hook: no bools
+
+// Hook fields:
+// module address
+// entity ID
+// hook type
+// if exec hook: hasPre, hasPost
+
+// Hook config is a packed representation of a hook function and flags for its configuration.
+// Layout:
+// 0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA________________________ // Address
+// 0x________________________________________BBBBBBBB________________ // Entity ID
+// 0x________________________________________________CC______________ // Type
+// 0x__________________________________________________DD____________ // exec hook flags
+//
+
+// Hook types:
+// 0x00 // Exec
+// 0x01 // Validation
```

Comment:
nit:
```suggestion
// 0x01 // Validation & Permission
```

Reply:
Permission hooks (exec hooks associated with a validation) are still represented as exec hooks in this format, because they have two bools for pre and post hook status.

Reply:
That right. Maybe we should add that in the comments.

Reply:
Will do